warehouse/query-core/src/main/java/datawave/query/jexl/visitors/SatisfactionVisitor.java

```diff
@@ -114,16 +114,32 @@ public Object visit(ASTEQNode node, Object data) {
     @Override
     public Object visit(ASTAndNode and, Object data) {
         QueryPropertyMarker.Instance instance = QueryPropertyMarker.findInstance(and);
-        // Recurse only if not delayed
-        if (!instance.isType(DELAYED)) {
-            if (instance.isNotAnyTypeOf(EXCEEDED_OR, EXCEEDED_VALUE)) {
-                and.childrenAccept(this, data);
+        if (instance.isAnyType() && instance.getSource() != null) {
+            switch (instance.getType()) {
+                case BOUNDED_RANGE:
+                case EXCEEDED_OR:
+                case EXCEEDED_VALUE:
+                    // the fields of the source node should be visited as a double lock safety check
+                    return null;
+                case DELAYED:
+                case EVALUATION_ONLY:
+                    isQueryFullySatisfied = false;
+                    return null;
+                case EXCEEDED_TERM:
+                    log.warn("cannot support exceeded term marker");
+                    isQueryFullySatisfied = false;
+                    return null;
+                case STRICT:
+                case LENIENT:
+                case INDEX_HOLE:
```

Comment:
index hole means cannot use global index, but it can use the field index.  I thing that one should set satisfied to false.

Reply:
also please verify the strict and lenient.  Those had to do with model expansion, not index lookup capabilities.